ui/src/shared/components/Gauge.js

```diff
@@ -0,0 +1,337 @@
+      ctx.lineCap = 'round'
+      ctx.strokeStyle = lineColor
+      ctx.moveTo(arcRadius, 0)
+      ctx.lineTo(arcRadius + tickSizeSmall, 0)
+      ctx.stroke()
+      ctx.closePath()
+      // Return to starting rotation
+      ctx.rotate(-lt * arcSmallIncrement)
+      ctx.rotate(-startDegree)
+    }
+  }
+
+  drawGaugeLabels = (
+    ctx,
+    xc,
+    yc,
+    radius,
+    gradientThickness,
+    minValue,
+    maxValue
+  ) => {
+    const {degree, lineCount, labelColor, labelFontSize} = GAUGE_SPECS
+
+    const incrementValue = (maxValue - minValue) / lineCount
+
+    const gaugeValues = []
+    for (let g = minValue; g < maxValue; g += incrementValue) {
+      const roundedValue = Math.round(g * 100) / 100
+      gaugeValues.push(roundedValue.toString())
+    }
+    gaugeValues.push((Math.round(maxValue * 100) / 100).toString())
+
+    const startDegree = degree * 135
+    const arcLength = Math.PI * 1.5
+    const arcIncrement = arcLength / lineCount
+
+    // Format labels text
+    ctx.font = `bold ${labelFontSize}px Helvetica`
+    ctx.fillStyle = labelColor
+    ctx.textBaseline = 'middle'
+    ctx.textAlign = 'right'
+    let labelRadius
+
+    for (let i = 0; i <= lineCount; i++) {
+      if (i === 3) {
+        ctx.textAlign = 'center'
+        labelRadius = radius + gradientThickness + 30
+      } else {
+        labelRadius = radius + gradientThickness + 23
+      }
+      if (i > 3) {
+        ctx.textAlign = 'left'
+      }
+      ctx.rotate(startDegree)
+      ctx.rotate(i * arcIncrement)
+      ctx.translate(labelRadius, 0)
+      ctx.rotate(i * -arcIncrement)
+      ctx.rotate(-startDegree)
+      ctx.fillText(gaugeValues[i], 0, 0)
+      ctx.rotate(startDegree)
+      ctx.rotate(i * arcIncrement)
+      ctx.translate(-labelRadius, 0)
+      ctx.rotate(i * -arcIncrement)
+      ctx.rotate(-startDegree)
+    }
+  }
+
+  drawGaugeValue = (ctx, radius, labelValueFontSize) => {
+    const {gaugePosition} = this.props
+    const {valueColor} = GAUGE_SPECS
+
+    ctx.font = `${labelValueFontSize}px Roboto`
+    ctx.fillStyle = valueColor
+    ctx.textBaseline = 'middle'
+    ctx.textAlign = 'center'
+
+    const textY = radius
+    ctx.fillText(gaugePosition.toString(), 0, textY)
+  }
+
+  drawNeedle = (ctx, radius, minValue, maxValue) => {
+    const {gaugePosition} = this.props
+    const {degree, needleColor0, needleColor1} = GAUGE_SPECS
+    const arcDistance = Math.PI * 1.5
+
+    const needleRotation = (gaugePosition - minValue) / (maxValue - minValue)
+
+    const needleGradient = ctx.createLinearGradient(0, -10, 0, radius)
+    needleGradient.addColorStop(0, needleColor0)
+    needleGradient.addColorStop(1, needleColor1)
+
+    // Starting position of needle is at minimum
+    ctx.rotate(degree * 45)
+    ctx.rotate(arcDistance * needleRotation)
+    ctx.beginPath()
+    ctx.fillStyle = needleGradient
+    ctx.arc(0, 0, 10, 0, Math.PI, true)
+    ctx.lineTo(0, radius)
+    ctx.lineTo(10, 0)
+    ctx.fill()
+  }
```

Comment:
Do you need to closePath here also?

Reply:
I opted to not close the path as that would connect the starting and ending points

![screen shot 2017-12-04 at 4 38 23 pm](https://user-images.githubusercontent.com/2433762/33583915-ad8af7f6-d911-11e7-84f5-57ddd5dd1f30.png)

Reply:
I believe `beginPath` resets the previous drawing instructions as well

Reply:
Mkay, sounds good, just happy to know you've put some thought into it. 👍